### PR TITLE
add test for multikeys

### PR DESCRIPTION
**Pull request by evanmcc**
support for https://github.com/helium/blockchain-core/pull/526

### CODE REVIEW

File: src/handlers/miner_dkg_handler.erl [hunk 1]
```diff
@@ -142,8 +142,6 @@ handle_message(BinMsg, Index, State=#state{n = N, t = T,
                                 true ->
                                     case State#state.done_called of
                                         false ->
-                                            %% this needs to be a call so we know the callback succeeded so we
```
Comment (Vagabond) [inline src/handlers/miner_dkg_handler.erl:145]: Why was this changed?

Comment (Vagabond) [reply src/handlers/miner_dkg_handler.erl:145]: So I took care of this over in #537, I guess we should merge that and then rebase this to use that?

File: test/miner_blockchain_SUITE.erl [hunk 2]
```diff
@@ -334,14 +338,160 @@ election_test(Config) ->
...
+    #{secret := Priv3, public := Pub3} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv4, public := Pub4} = libp2p_crypto:generate_keys(ecc_compact),
+    #{secret := Priv5, public := Pub5} = libp2p_crypto:generate_keys(ecc_compact),
+    BinPub1 = libp2p_crypto:pubkey_to_bin(Pub1),
+    BinPub2 = libp2p_crypto:pubkey_to_bin(Pub2),
+    BinPub3 = libp2p_crypto:pubkey_to_bin(Pub3),
+    BinPub4 = libp2p_crypto:pubkey_to_bin(Pub4),
+    BinPub5 = libp2p_crypto:pubkey_to_bin(Pub5),
+
+    Txn6 = vars(#{?use_multi_keys => true}, 2, Priv),
```
Comment (Vagabond) [inline test/miner_blockchain_SUITE.erl:372]: Why are these being done in discrete var updates?

File: test/miner_blockchain_SUITE.erl [hunk 3]
```diff
@@ -334,14 +338,160 @@ election_test(Config) ->
...
+               #{}, 3,
+               #{multi_keys => [BinPub1, BinPub2, BinPub3, BinPub4, BinPub5]}),
+    Proofs7 = [blockchain_txn_vars_v1:create_proof(P, Txn7_0)
+               || P <- [Priv1, Priv2, Priv3, Priv4, Priv5]],
+    Txn7_1 = blockchain_txn_vars_v1:multi_key_proofs(Txn7_0, Proofs7),
+    Proof7 = blockchain_txn_vars_v1:create_proof(Priv, Txn7_1),
+    Txn7 = blockchain_txn_vars_v1:proof(Txn7_1, Proof7),
+    _ = [ok = ct_rpc:call(M, blockchain_worker, submit_txn, [Txn7]) || M <- Miners],
+
+    ct:pal("transitioned to multikey"),
```
Comment (Vagabond) [inline test/miner_blockchain_SUITE.erl:387]: Should we use wait_for_chain_var_update here too?

File: test/miner_blockchain_SUITE.erl (hunk 2 above)
Comment (evanmcc) [reply test/miner_blockchain_SUITE.erl:372]: to make sure that that transition works safely, just in case it happens.  it's done the other way, as an atomic update, in the election test.

Comment (evanmcc) [reply test/miner_blockchain_SUITE.erl:372]: oops, that's not true, will change this one.

File: test/miner_blockchain_SUITE.erl (hunk 3 above)
Comment (evanmcc) [reply test/miner_blockchain_SUITE.erl:387]: yeah, fixing

**Comment by Vagabond** [comment]
```
miner_blockchain_SUITE:master_key_test failed on line 737
Reason: {badmatch,false}
```

**Review by Vagabond — APPROVED** [review]
